### DISCUSSION TITLE
Umi 如何使用内联样式表

### DISCUSSION THREAD

**Discussion by Echonessy** [umijs/umi#11931] [Q&A]
import inlines from '@arco-themes/react-aim-theme/index.less?inline';
export default () => {
  return (
    <div>
     <style>{inlines}</style>
      <p>2342</p>
    </div>
  );
};

**Comment by fz6m** [comment]
目前不支持这种 import 的方式得到源内容，如有懒加载需求，可以手动 `React.lazy(() => import('./path/to/component'))` 拆包。

也可以通过修改 webpack 配置来实现含有特定 resource query 的不同导入行为，具体请自行探究。

**Reply by Echonessy** [reply]
那我考虑下把这部分拆分出来 发布到npm使用了。现在的应用场景就是 基于mjml的编辑器集成到umi项目里，但是mjml又需要大量的内联样式表，所以融合到一起全是报错，调试了一天也没发现什么解决方法，webpack配置似乎也起不到什么作用

**Reply by fz6m** [reply]
你用 `<style></style>` 这种内联样式和直接导入这个样式文件变成全局样式没有区别的：

```ts
import '/path/to/style.css' // 导入为全局样式
```

把这部分编辑器的组件单独拆出来然后用 `React.lazy()` 和 `Suspense` 配合懒加载就可以了，这样这部分 css 样式就在需要的时候才会加载，拆出来第三方包甚至也不需要。

**Comment by Echonessy** [comment]


**Reply by Echonessy** [reply]
那我考虑下把这部分拆分出来 发布到npm使用了。现在的应用场景就是 基于mjml的编辑器集成到umi项目里，但是mjml又需要大量的内联样式表，所以融合到一起全是报错，调试了一天也没发现什么解决方法，webpack配置似乎也起不到什么作用
